lib/src/puppetdb-cli.cc

```diff
@@ -50,8 +56,20 @@ parse_config() {
 
 string
 pdb_server_url(const json::JsonContainer& config) {
-    const auto server_urls = config.get< vector<string> >("server_urls");
-    return server_urls.size() ? server_urls[0] : "http://127.0.0.1:8080";
+    const auto server_urls_datatype = config.type("server_urls");
+    if (server_urls_datatype == json::DataType::Array) {
+        const auto server_urls = config.get< vector<string> >("server_urls");
+        return server_urls.size() ? server_urls[0] : "http://127.0.0.1:8080";
```

Comment:
It seems like we need this defaulting here if the user has provided a config that does not include a server_url. Can we just alter the config code above to add in the server_url if not present? Then this can go away.